.buildkite/e2e/reporter/main.go

```diff
@@ -195,14 +208,41 @@ func sortByName(tests []junit.Test) {
 	})
 }
 
-func trimError(err error, bytes int) error {
+func truncateError(err error, length int) error {
 	msg := []byte(err.Error())
-	if len(msg) > bytes {
-		return errors.New(string(msg[0:bytes]))
+	if len(msg) > length {
+		return errors.New(string(msg[0 : length-1]))
 	}
 	return err
 }
 
+func truncateNotifyMessage(yamlDef []byte, length int) ([]byte, error) {
+	var pipeline struct {
+		Steps []struct {
+			Notify []struct {
+				Slack struct {
+					Message string `yaml:"message"`
+				} `yaml:"slack"`
+			} `yaml:"notify"`
+		} `yaml:"steps"`
+	}
+	err := yaml.Unmarshal(yamlDef, &pipeline)
+	if err != nil {
+		return nil, nil
+	}
+	for i, s := range pipeline.Steps {
+		for j, n := range s.Notify {
+			// truncate message and replace last 3 chars by '...'
+			pipeline.Steps[i].Notify[j].Slack.Message = n.Slack.Message[0:length-1-3] + "..."
```

Comment:
Isn't this going to fail with `panic: runtime error: slice bounds out of range [:3000] with capacity x` when the message isn't at least of length 3000? A quick test in go playground seems to verify this.

Reply:
You are right, it needs a length check before truncation.